aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -113,13 +144,21 @@ private Completable createHydrationTask(
                     // So that when we save them, the references will exist.
                     .sorted(modelWithMetadataComparator::compare)
                     // For each ModelWithMetadata, merge it into the local store.
-                    .flatMapCompletable(merger::merge)
+                    .flatMapCompletable(modelWithMetadata ->
+                        merger.merge(modelWithMetadata, metricsAccumulator::increment)
+                    )
                     .toSingle(() -> lastSyncTime.exists() ? SyncType.DELTA : SyncType.BASE);
             })
             .flatMapCompletable(syncType -> {
-                return SyncType.DELTA.equals(syncType) ?
+                Completable syncTimeSaveCompletable = SyncType.DELTA.equals(syncType) ?
                     syncTimeRegistry.saveLastDeltaSyncTime(modelClass, SyncTime.now()) :
                     syncTimeRegistry.saveLastBaseSyncTime(modelClass, SyncTime.now());
+                return syncTimeSaveCompletable.andThen(Completable.fromAction(() -> {
+                    metricsAccumulator
+                        .toModelSyncedEvent(syncType)
+                        .toHubEvent()
+                        .publish(HubChannel.DATASTORE, Amplify.Hub);
```

Comment:
What implications does this have on testability, if you inline `Amplify.Hub`? Should we instead snake a `hub` instance in through the constructor? That way, you could provide a mock in test, if you needed to?

I forget why we put this in the publish arg... it seemed like a good idea at the time?

Reply:
The original intent was to not have `Amplify.Hub` references inside the `HubEvent` class. Let me ponder over this one for a bit.

Reply:
I may have simply been wrong on this. It does feel like encapsulating it HubEvent is cleaner than stating it at random call sites like this, doesn't it?

Reply:
I've gone back and changed this a bit. I see `HubEvent` more as a POJO rather than something that has behavior. I changed the call sites back to:
```java 
Amplify.Hub.publish(HubChannel.DATASTORE, <eventDataPojo>.toHubEvent());
```